ui/app/clinical/consultation/directives/consultationConfirmOnExit.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+angular.module('bahmni.clinical')
+    .directive('consultationConfirmOnExit', ['clinicalAppConfigService', 'messagingService', 'spinner', '$state',
+        function (clinicalAppConfigService, messagingService, spinner, $state) {
+            return {
+                link: function ($scope, elem, attrs) {
+                    window.onbeforeunload = function () {
+                        console.log("change");
+                    };
+                    $scope.$on('$stateChangeStart', function (event, next, current) {
+                        var noOfMedications = 0;
+                        var navigating = next.url.split("/")[1];
+                        var allConsultationBoards = clinicalAppConfigService.getAllConsultationBoards();
+                        var outOfConsultationBoard = true;
+                        if ($scope.consultation) {
+                            if ($scope.consultation.orders) {
+                                if ($scope.consultation.orders.length !== $scope.consultation.investigations.length) {
+                                    $state.dirtyConsultationForm = true;
+                                }
+                            }
+                            if ($scope.consultation.newlyAddedTabTreatments && $scope.tabConfigName) {
+                                noOfMedications = $scope.consultation.newlyAddedTabTreatments[$scope.tabConfigName].treatments.length;
+                            }
+                        }
+
+                        var isConsultationFormDirty = noOfMedications > 0 || ($scope[attrs.name] && $scope[attrs.name].$dirty);
+                        if (isConsultationFormDirty) {
+                            $state.dirtyConsultationForm = true;
+                        }
+                        allConsultationBoards.map(function (board) {
```

Comment:
Please do not use map to loop through an array, a map is generally used to return a new array using its callback function, You an use a simple forEach or for loop in such cases

Reply:
Updated